marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mapping/mapping-lib.sjs

```diff
@@ -0,0 +1,38 @@
+/**
+ Copyright (c) 2021 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+const hubUtils = require("/data-hub/5/impl/hub-utils.sjs");
+
+function extractFriendlyErrorMessage(e){
+  let errorMessage;
+  switch(e.name){
+    case "XDMP-UNDFUN":
+      errorMessage = `Unable to find function: '${e.data[0]}'. Cause: Either the function does not exist or the wrong number of arguments were specified.`;
+      break;
+    default:
+      errorMessage = null;
+  }
+  return errorMessage;
+}
+function extractErrorMessageForMappingUI(e){
+  let errorMessage = extractFriendlyErrorMessage(e);
+  return errorMessage ? errorMessage : hubUtils.getErrorMessage(e);
+}
+
+module.exports = {
+  extractErrorMessageForMappingUI,
```

Comment:
This shouldn't have "UI" in it, because this functionality will be used regardless of how the step is run - e.g. if you run the step via Gradle and there's an error, it will show up in stepOutput. 

Perhaps just have one function here with a "boolean fallbackToGenericErrorMessage" that controls whether or not hubUtils.getErrorMessage is used.

Reply:
1. extractErrorMessageForMappingUI() will be used exclusively in HC mapping UI when user clicks on "Test" button. The fallback when the error is not the one mentioned in the jira stories here is hubUtils.getErrorMessage(e) (as it was earlier)
2. extractFriendlyErrorMessage() will be used when you run the step from gradle or from HC run tile. The fallback when the error is not the one mentioned in the jira stories here is throw Error(e) (as it was earlier)